.gitignore

```diff
@@ -1,24 +1,79 @@
-.DS_Store
-*.pyc
-*~
-*swp
-TFC Build/
-bin/
-.gradle/
+# Created by https://www.gitignore.io
+# Following settings: Gradle, intellij, eclipse
+### Gradle ###
+.gradle
 build/
+
+# Ignore Gradle GUI config
+gradle-app.setting
+
+# Avoid ignoring Gradle wrapper jar file (.jar files are usually ignored)
+!gradle-wrapper.jar
+
+
+### Intellij ###
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm
+
+*.iml
+
+## Directory-based project format:
+.idea/
+
+## File-based project format:
+*.ipr
+*.iws
+
+## Plugin-specific files:
+
+# IntelliJ
+/out/
+
+# mpeltonen/sbt-idea plugin
+.idea_modules/
+
+# JIRA plugin
+atlassian-ide-plugin.xml
+
+# Crashlytics plugin (for Android Studio and IntelliJ)
+com_crashlytics_export_strings.xml
+crashlytics.properties
+crashlytics-build.properties
+
+
+### Eclipse ###
+*.pydevproject
+.metadata
+.gradle
+bin/
+tmp/
+*.tmp
+*.bak
+*.swp
+*~.nib
+local.properties
 .settings/
-logs/
-crash-reports/
-config/
-saves/
-screenshots/
+.loadpath
+
+# Eclipse Core
 .project
+
+# External tool builders
+.externalToolBuilders/
+
+# Locally stored "Eclipse launch configurations"
+*.launch
+
+# CDT-specific
+.cproject
+
+# JDT-specific (Eclipse Java Development Tools)
 .classpath
-options.txt
-out
-*.iws
-*.ipr
-*.iml
-mods
-asm
-*Thumbs.db

```

Comment:
This line was there for my desktop, due to some issues with transferring the files, and should not have been removed.